src/Tinfoil/Key.hs

```diff
@@ -0,0 +1,18 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Tinfoil.Key(
+    genSymmetricKey256
+) where
+
+import           P
+
+import           System.IO (IO)
+
+import           Tinfoil.Data.Key
+import           Tinfoil.Data.Random
+import           Tinfoil.Random
+
+-- | Generate a 256-bit symmetric cryptographic key.
+genSymmetricKey256 :: IO SymmetricKey
+genSymmetricKey256 = fmap (SymmetricKey . unEntropy) $ entropy 32
```

Comment:
Yeah, should compile to the same code, but its `fmap (SymmetricKey . unEntropy) $` vs `(SymmetricKey . unEntropy) <$>`. I don't have a strong opinion on this. I've been using hlint on my own projects and this is something hlint would complain about.